drake/solvers/trajectoryOptimization/InverseKinematicsTrajectory.m

```diff
@@ -133,21 +133,17 @@
         end
         if(isa(varargin{i},'SingleTimeKinematicConstraint'))
           for j = t_start:obj.nT
-            if(varargin{i}.isTimeValid(obj.t_knot(j)))
-              cnstr = varargin{i}.generateConstraint(obj.t_knot(j));
-              obj = obj.addKinematicConstraint(cnstr{1},j);
-            end
+            cnstr = varargin{i}.generateConstraint(obj.t_knot(j));
```

Comment:
For a SingleTimeKinematicConstraint, now that it does not depend on time, its `generateConstraint` function should not take `t` as the input argument.